src/atomate2/abinit/jobs/base.py

```diff
@@ -0,0 +1,815 @@
+"""Definition of base ABINIT job maker."""
+
+from __future__ import annotations
+
+import errno
+import json
+import logging
+import os
+import shutil
+import time
+from dataclasses import dataclass, field
+from typing import Any, ClassVar, Dict, List, Optional, Sequence, Union
+
+import jobflow
+import pseudo_dojo
+from abipy.electrons.gsr import GsrFile
+from abipy.flowtk import events
+from abipy.flowtk.events import AbinitEvent
+from abipy.flowtk.utils import Directory, File, irdvars_for_ext
+from jobflow import Maker, Response, job
+from monty.inspect import all_subclasses
+from monty.json import MontyEncoder
+from monty.string import is_string
+from pymatgen.core.structure import Structure
+from pymatgen.io.abinit.pseudos import PseudoTable
+
+from atomate2.abinit.run import run_abinit
+from atomate2.abinit.schemas.core import AbinitJobSummary
+from atomate2.abinit.sets.base import AbinitInputSet, AbinitInputSetGenerator
+from atomate2.abinit.utils.common import (
+    HISTORY_JSON,
+    INDATA_PREFIX,
+    INDIR_NAME,
+    INPUT_FILE_NAME,
+    LOG_FILE_NAME,
+    MPIABORTFILE,
+    OUTDATA_PREFIX,
+    OUTDIR_NAME,
+    OUTPUT_FILE_NAME,
+    STDERR_FILE_NAME,
+    TMPDIR_NAME,
+    InitializationError,
+    PostProcessError,
+    UnconvergedError,
+)
+from atomate2.abinit.utils.history import JobHistory
+from atomate2.abinit.utils.settings import AbinitAtomateSettings, get_abipy_manager
+
+logger = logging.getLogger(__name__)
+
+__all__ = ["BaseAbinitMaker"]
+
+
+def as_event_class(event_string):
+    """Convert event string into a subclass of AbinitEvent.
+
+    The string can be the class name or the YAML tag.
+    """
+    if is_string(event_string):
+        for c in all_subclasses(AbinitEvent):
+            if c.__name__ == event_string or c.yaml_tag == event_string:
+                return c
+        raise ValueError(f"Cannot find event class associated to {event_string}.")
+    raise ValueError(
+        f"Cannot convert event_string of type {type(event_string)}. Should be a string."
+    )
+
+
+@dataclass
+class BaseAbinitMaker(Maker):
+    """
+    Base ABINIT job maker.
+
+    Parameters
+    ----------
+    name : str
+        The job name.
+    pseudos : list of str, PseudoTable
+        The pseudopotentials to use.
+    """
+
+    calc_type: str
```

Comment:
Is it possible to have calc type defined by the input set?

Reply:
Done.